pages/transactions/[id].js

```diff
@@ -73,6 +75,14 @@ export default function TransactionDetailsPage({
 
   return (
     <main>
+      <StyledLink href={"/settings"}>
```

Comment:
Each interactive element that includes an image or an icon instead of textcontent should have an aria-label.
By the way: Since you render this pattern at different places you can consider making it a component.